src/palimpzest/operators/operators.py

```diff
@@ -167,38 +165,29 @@ def _createPhysicalPlans(self, max: int=10, shouldProfile: bool=False) -> Physic
 
         return physicalPlans
 
-    def createPhysicalPlanCandidates(self, sampler: Sampler=None, shouldProfile: bool=False) -> List[PhysicalPlan]:
+    def createPhysicalPlanCandidates(self, cost_estimate_sample_data: List[Dict[str, Any]]=None, shouldProfile: bool=False) -> List[PhysicalPlan]:
         """Return a set of physical trees of operators."""
-        start_time = time.time()
-
         # create set of logical plans (e.g. consider different filter/join orderings)
         logicalPlans = self._createLogicalPlans()
-        t_logical = time.time()
-        # print(f"Time to create logical plans: {t_logical - start_time:.2f}")
 
         # iterate through logical plans and evaluate multiple physical plans
         physicalPlans = [
             physicalPlan
             for logicalPlan in logicalPlans
             for physicalPlan in logicalPlan._createPhysicalPlans(shouldProfile=shouldProfile)
         ]
-        t_physical = time.time()
-        # print(f"Time to create physical plans: {t_physical - t_logical:.2f}")
 
         # estimate the cost (in terms of USD, latency, throughput, etc.) for each plan
         plans = []
         for physicalPlan in physicalPlans:
-            planCost = physicalPlan.estimateCost()
+            planCost = physicalPlan.estimateCost(cost_estimate_sample_data=cost_estimate_sample_data)
```

Comment:
If the dictionary is not provided, is the intention that estimateCost() use historical data?

Reply:
If the list[dict] is not provided then it should use our estimates w/zero prior information (i.e., just using constants for things like selectivity, read time, etc.)